kernel/device.c

```diff
@@ -84,17 +88,22 @@ const struct device *z_impl_device_get_binding(const char *name)
 	 * with CONFIG_* macros), only cheap pointer comparisons will be
 	 * performed. Reserve string comparisons for a fallback.
 	 */
+	LOG_DBG("device_get_binding(%s)", name);
 	for (dev = __device_start; dev != __device_end; dev++) {
 		if (z_device_ready(dev) && (dev->name == name)) {
+			LOG_DBG("   - found");
 			return dev;
 		}
 	}
 
 	for (dev = __device_start; dev != __device_end; dev++) {
+		LOG_DBG("   - %s, ready=%d", dev->name, z_device_ready(dev));
```

Comment:
Printing all the devices sounds unnecessary, as the "not found" is printed at the end anyway.

Reply:
Because I want to see why it was not found. It could be because the thing I am looking for does not exist, or because it is not ready.